src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/Frame.cs

```diff
@@ -374,6 +377,11 @@ public void Reset()
 
             _responseBytesWritten = 0;
             _requestCount++;
+
+            // TODO: probably not the right place to complete
+            RequestBodyPipe.Reader.Complete();
+            RequestBodyPipe.Writer.Complete();
+            RequestBodyPipe.Reset();
```

Comment:
`Reset()` fails if the reader and writer are not completed:

```
System.InvalidOperationException : Both reader and writer need to be completed to be able to reset
Stack Trace:
     at Microsoft.AspNetCore.Server.Kestrel.Internal.System.IO.Pipelines.Pipe.Reset()
     at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Frame.Reset()
     at Microsoft.AspNetCore.Server.Kestrel.Core.Tests.FrameTests..ctor()
```

and we call `Frame.Reset()` essentially before anything else: https://github.com/aspnet/KestrelHttpServer/blob/1a706113d9f75999d9642ccc05b41593f7c0aa50/src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameOfT.cs#L39